examples/grpc-service/src/main/java/example/armeria/grpc/HelloServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package example.armeria.grpc;
+
+import java.time.Duration;
+import java.util.ArrayList;
+
+import example.armeria.grpc.Hello.HelloReply;
+import example.armeria.grpc.Hello.HelloRequest;
+import example.armeria.grpc.HelloServiceGrpc.HelloServiceImplBase;
+import io.grpc.stub.StreamObserver;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class HelloServiceImpl extends HelloServiceImplBase {
+
+    /**
+     * Sends an {@link HelloReply} immediately when receiving a request.
+     */
+    @Override
+    public void hello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        responseObserver.onNext(buildReply(toMessage(request.getName())));
+        responseObserver.onCompleted();
+    }
+
+    /**
+     * Sends an {@link HelloReply} 3 seconds later after receiving a request.
+     */
+    @Override
+    public void lazyHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        // Respond 3 seconds later.
+        Mono.delay(Duration.ofSeconds(3))
+            .subscribe(unused -> responseObserver.onNext(buildReply(toMessage(request.getName()))),
+                       responseObserver::onError, responseObserver::onCompleted);
+    }
+
+    /**
+     * Sends 5 {@link HelloReply} responses when receiving a request.
+     */
+    @Override
+    public void lotsOfReplies(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        Flux.interval(Duration.ofSeconds(1))
```

Comment:
This would be less elegant without the reactor, but I might still lean towards making the example use as little non-gRPC / non-armeria concepts as possible to make it easier to read.

Reply:
I think the callbacks may or may not be called with context mounted, which isn't good practice. Should this example mention the pattern to make sure the `subscribe` callbacks are called with the context aware loop? I think it's `subscribeOn` but I'm not super-familiar with these APIs.

Reply:
Thanks! Good point. Will make a fix for that. :-)